app/_src/mesh/features/meshopa.md

```diff
@@ -0,0 +1,668 @@
+---
+title: MeshOPA (beta) - OPA Policy Integration
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [OPA Policy](../opa).
+{% endwarning %}
+
+## MeshOPA policy plugin
+
+{{site.mesh_product_name}} integrates the [Open Policy Agent (OPA)](https://www.openpolicyagent.org/) to provide access control for your services.
+
+The agent is included in the data plane proxy sidecar, instead of the more common deployment as a separate sidecar.
+
+When `MeshOPA` policy is applied, the control plane configures:
+
+- The embedded policy agent, with the specified policy
+- Envoy, to use [External Authorization](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto) that points to the embedded policy agent
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅        | ❌  | ❌   |
+| MeshSubset        | ✅        | ❌  | ❌   |
+| MeshService       | ✅        | ❌  | ❌   |
+| MeshServiceSubset | ✅        | ❌  | ❌   |
+| MeshGatewayRoute  | ❌        | ❌  | ❌   |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+To apply a policy with MeshOPA:
+
+- Specify the group of data plane proxies to apply the policy to with the `targetRef` property.
+- Provide the list of policies<a href="#footnote-1"><sup id="note-return-1">1</sup></a> with the `appendPolicies` property. Policies are defined in the [Rego language](https://www.openpolicyagent.org/docs/latest/policy-language/).
+- Optionally provide custom configuration for the policy agent.
+
+<span id="footnote-1"><a href="#note-return-1">1</a></span>: due to a [bug in OPA](https://github.com/open-policy-agent/opa/issues/5595) we currently support only one policy. 
+Be aware that due to [configuration merging](/mesh/{{page.kong_version}}/policies/targetref/#merging-configuration) you can end up with more than one entry in `appendPolicies`.
+
+You must also specify the HTTP protocol in your mesh configuration:
+
+{% navtabs %}
+{% navtab Kubernetes %}
+
+Add the HTTP protocol annotation to the Kubernetes Service configuration, with the general syntax `<port>.service.kuma.io/protocol`.
```

Comment:
is this really needed? I think it is sufficient if you specify `appProtocol` on k8s, and you don't need: 
```
annotations:
    8080.service.kuma.io/protocol: http
```

Reply:
I'm not sure, this is how it's in old OPA policy docs - https://github.com/Kong/docs.konghq.com/blame/02ae7fd2dd8c64582a76d0dd5e6dee79d09db7d4/app/_src/mesh/features/opa.md#L31

Reply:
Figure out that his is a deprecated way of hinting a protocol, using `appProtocol` now.